tools/debugging/channels_with_minimum_balance.py

```diff
@@ -0,0 +1,273 @@
+
+            if not is_successful_reponse(response):
+                raise RuntimeError(f"An error ocurrent while opening channel {channel_open}")
+
+        else:
+            deposit = ChannelDeposit(
+                channel_open.token_address,
+                channel_open.participant2,
+                channel_open.endpoint1,
+                channel_open.minimum_capacity1,
+            )
+            channel_deposit_if_necessary(channel, deposit)
+
+        # A deposit only makes sense after the channel is opened.
+        deposit = ChannelDeposit(
+            channel_open.token_address,
+            channel_open.participant1,
+            channel_open.endpoint2,
+            channel_open.minimum_capacity2,
+        )
+
+        log.info(f"Queueing {deposit}")
+        target_to_depositqueue[(channel_open.token_address, channel_open.participant2)].put(
+            deposit
+        )
+
+
+def channel_deposit_with_the_same_node_and_token_network(deposit_queue: JoinableQueue) -> None:
+    """Because of how the ERC20 standard is defined, two concurrent approve
+    calls overwrite each other.
+
+    Additionally, to prevent a node from trying to deposit more tokens than it
+    has, and by consequence sending an unnecessary transaction, a lock is used.
+    (e.g.: When two transactions that are individually valid, but together use
+    more than the account's balance). This has the side effect of forbiding
+    concurrent deposits on the same token network. (Issue #5447)
+    """
+    while True:
+        deposit = deposit_queue.get()
+
+        channel = channel_details(deposit.endpoint, deposit.token_address, deposit.partner)
+        if channel is None:
+            raise RuntimeError(f"Channel does not exist! {deposit}")
+        channel_deposit_if_necessary(channel, deposit)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("config")
+
+    args = parser.parse_args()
+
+    with open(args.config, "r") as handler:
+        config = json.load(handler)
+
+    # validate the endpoints
+    node_to_endpoint = dict()
+    node_to_address = dict()
+    for node_name, node_info in config["nodes"].items():
+        if urlsplit(node_info["endpoint"]).scheme == "":
+            raise ValueError("'endpoint' must have the protocol defined")
+
+        url_deposit = f"{node_info['endpoint']}/api/{API_VERSION}/address"
+        result = requests.get(url_deposit).json()
+
+        if result["our_address"] != node_info["address"]:
+            raise ValueError(
+                f"Address mismatch, configuration {node_info['address']}, "
+                f"API response {result['our_address']}"
+            )
+
+        node_to_endpoint[node_name] = node_info["endpoint"]
+        node_to_address[node_name] = node_info["address"]
+
+    queue_per_node: Dict[str, List[ChannelNew]] = defaultdict(list)
+    target_to_depositqueue: Dict[Tuple[str, str], JoinableQueue] = dict()
```

Comment:
These two could be named more consistently. Maybe use `node_to_newchannelqueue` or `node_to_openqueue` instead of `queue_per_node`?